src/useMediaQuery.js

```diff
@@ -0,0 +1,25 @@
+export function useMediaQuery(query) {
+  const [matches, setMatches] = React.useState(null)
+
+  React.useEffect(
+    () => {
+      const mql = window.matchMedia(query)
+      setMatches(mql.matches)
+
+      mql.addEventListener
+        ? mql.addEventListener('change', handleChange)
+        : mql.addListener(handleChange)
```

Comment:
Je zou kunnen zeggen dat, wanneer IE ondersteund moet worden (redelijk onwaarschijnlijk voor nieuwe sites) iemand de polyfill moet gebruiken:

`MediaQueryList.prototype.addEventListener,MediaQueryList.prototype.removeEventListener`

Support voor addEventListener is best wel goed: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList#browser_compatibility

Reply:
Right, dit is het missende puzzel stukje. Ik ben altijd van `matchMedia` als polyfill uitgegaan. Die voegt wel `addListener` toe, maar niet `addEventListener`. Ik dacht dat dat een fout was, maar deze bestaan dus los. Thanks!